Framework/PythonInterface/plugins/algorithms/VesuvioTOFFit.py

```diff
@@ -112,17 +111,22 @@ def _run_fit_impl(self, data_ws, workspace_index, fit_options, simulation=False)
             function_str = fit_options.create_function_str()
             constraints = fit_options.create_constraints_str()
             ties = fit_options.create_ties_str()
-            results = self._do_fit(function_str, data_ws, workspace_index, constraints, ties,
-                                   max_iter=self.getProperty("MaxIterations").value)
+
+            chi_square, params, fitted_data = self._do_fit(function_str, data_ws, workspace_index,
+                                                       constraints, ties,
+                                                       max_iter=self.getProperty("MaxIterations").value)
 
             # Run second time using standard CompositeFunction & no constraints matrix to
             # calculate correct reduced chisq
-            param_values = TableWorkspaceDictionaryFacade(results[1])
+            param_values = TableWorkspaceDictionaryFacade(params)
 
         function_str = fit_options.create_function_str(param_values)
         max_iter = 0 if simulation else 1
-        return self._do_fit(function_str, data_ws, workspace_index, constraints,
-                            ties, max_iter=max_iter)
+        reduced_chi_square, _, _= self._do_fit(function_str, data_ws,
```

Comment:
I set up the fit in the fit wizard (which gave the expected value for Chi-squared) and showed it to Roman. The `CalculateChiSquared` algorithm wasn't giving the same result and Roman suggested this was to do with fits attempt to optimise the result/params. I thought that in this case it might be best to ensure we still get the results we expect rather than potentially changing the values